scilpy/utils/streamlines.py

```diff
@@ -148,45 +147,43 @@ def warp_tractogram(streamlines, transfo, deformation_data, source):
 
     # Because of duplication, an iteration over chunks of points is necessary
     # for a big dataset (especially if not compressed)
+    streamlines = ArraySequence(streamlines)
     nb_points = len(streamlines._data)
-    current_position = 0
+    cur_position = 0
     chunk_size = 1000000
     nb_iteration = int(np.ceil(nb_points/chunk_size))
     inv_transfo = np.linalg.inv(transfo)
 
     while nb_iteration > 0:
-        max_position = min(current_position + chunk_size, nb_points)
-        streamline = streamlines._data[current_position:max_position]
+        max_position = min(cur_position + chunk_size, nb_points)
+        points = streamlines._data[cur_position:max_position]
 
         # To access the deformation information, we need to go in voxel space
-        streamline_vox = transform_streamlines(streamline,
-                                               inv_transfo)
+        # No need for corner shift since we are doing interpolation
+        cur_points_vox = np.array(transform_streamlines(points,
+                                                        inv_transfo)).T
 
-        current_streamline_vox = np.array(streamline_vox).T
-        current_streamline_vox_list = current_streamline_vox.tolist()
-
-        x_def = ndimage.map_coordinates(deformation_data[..., 0],
-                                        current_streamline_vox_list, order=1)
-        y_def = ndimage.map_coordinates(deformation_data[..., 1],
-                                        current_streamline_vox_list, order=1)
-        z_def = ndimage.map_coordinates(deformation_data[..., 2],
-                                        current_streamline_vox_list, order=1)
+        x_def = map_coordinates(deformation_data[..., 0],
+                                cur_points_vox.tolist(), order=1)
+        y_def = map_coordinates(deformation_data[..., 1],
+                                cur_points_vox.tolist(), order=1)
+        z_def = map_coordinates(deformation_data[..., 2],
+                                cur_points_vox.tolist(), order=1)
 
         # ITK is in LPS and nibabel is in RAS, a flip is necessary for ANTs
-        final_streamline = np.array([flip[0]*x_def,
-                                     flip[1]*y_def,
-                                     flip[2]*z_def])
+        final_points = np.array([flip[0]*x_def, flip[1]*y_def, flip[2]*z_def])
 
-        # The deformation obtained is in worldSpace
+        # The Ants deformation is relative to world space
         if source == 'ants':
-            final_streamline += np.array(streamline).T
+            final_points += np.array(points).T
```

Comment:
In this case, we sent the streamlines to VOX space for nothing...

Reply:
We actually sent them to VOX space to get the deformation. But we simply apply the deformation to world space streamlines.
The problem with ants is that we need streamlines in both space at the same time.

Reply:
Oh, I see, sorry about that